src/provider/states/running.rs

```diff
@@ -35,16 +36,44 @@ impl State<PodState> for Running {
         _pod: &Pod,
     ) -> Transition<PodState> {
         loop {
-            tokio::select! {
-                _ = tokio::time::delay_for(std::time::Duration::from_secs(10))  => {
-                    trace!("Checking if service {} is still running.", &pod_state.service_name);
+            tokio::time::delay_for(Duration::from_secs(10)).await;
+            trace!(
+                "Checking if service {} is still running.",
+                &pod_state.service_name
+            );
+            if let Some(systemd_units) = &pod_state.service_units {
```

Comment:
If you want to reduce nesting you could return early instead of nesting it all here